### PR TITLE
Entrypoint to check for HTTPS_PORT if http_port is disabled.

### PR DESCRIPTION
**Pull request by HeenaBansal2009**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Update entrypoint.sh to ping HTTPS_port if http is not enabled in configuration.

### CODE REVIEW

File: docker/server/entrypoint.sh [hunk 1]
```diff
@@ -115,7 +117,7 @@ if [ -n "$(ls /docker-entrypoint-initdb.d/)" ] || [ -n "$CLICKHOUSE_DB" ]; then
     # check if clickhouse is ready to accept connections
     # will try to send ping clickhouse via http_port (max 12 retries by default, with 1 sec timeout and 1 sec delay between retries)
     tries=${CLICKHOUSE_INIT_TIMEOUT:-12}
-    while ! wget --spider -T 1 -q "http://127.0.0.1:$HTTP_PORT/ping" 2>/dev/null; do
+    while ! wget --spider -T 1 -q "http://127.0.0.1:$PORT/ping" 2>/dev/null; do
```
Comment (alexey-milovidov) [inline docker/server/entrypoint.sh:120]: This is not enough, if you are using https, you should specify https as the URL schema.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

File: docker/server/entrypoint.sh (hunk 1 above)
Comment (HeenaBansal2009) [reply docker/server/entrypoint.sh:120]: I'll update the endpoint URL  as https://127.0.0.1:$PORT/ping.

**Comment by alexey-milovidov** [comment]
Now it will prefer https for the health check. But there can be one complication.
Sometimes clickhouse-server is configured with self-signed certificate (or a certificate with internal CA) which is not registered in the system (or inside this Docker container). If wget will check it and refuse to connect, the health check will fail.

Let's:
- make HTTP preferred if it's available;
- add `--no-check-certificate` flag to wget.

File: docker/server/entrypoint.sh [hunk 2]
```diff
@@ -107,6 +107,13 @@ fi
 if [ -n "$(ls /docker-entrypoint-initdb.d/)" ] || [ -n "$CLICKHOUSE_DB" ]; then
     # port is needed to check if clickhouse-server is ready for connections
     HTTP_PORT="$(clickhouse extract-from-config --config-file "$CLICKHOUSE_CONFIG" --key=http_port)"
+    HTTPS_PORT="$(clickhouse extract-from-config --config-file "$CLICKHOUSE_CONFIG" --key=https_port)"
+    
+    if [ -n "$HTTP_PORT" ]; then
+        URL="https://127.0.0.1:$HTTPS_PORT/ping"
```
Comment (larryluogit) [inline docker/server/entrypoint.sh:113]: Do you mean HTTP here?

**Review by alexey-milovidov — APPROVED** [review]

**Comment by HeenaBansal2009** [comment]
@alexey-milovidov , Github actions are failing at CheckLabels. Please` add can be tested ` label.

**Comment by HeenaBansal2009** [comment]
@alexey-milovidov , I see that github actions are not triggered for this PR. I am not sure if GitHub actions are required for this PR to be merged ?
Please advice.

**Comment by HeenaBansal2009** [comment]
@alexey-milovidov ,None of the failures seems related to PR. 
The stress test is failing with below error and doesn't seems related.
`2022.08.17 12:19:32.173899 [ 981504 ] {} <Error> MutateFromLogEntryTask: virtual bool DB::ReplicatedMergeMutateTaskBase::executeStep(): Code: 47. DB::Exception: Missing columns: 'value4' while processing query: 'value4, value4, key, value0, value2', required columns: 'value4' 'key' 'value0' 'value2', maybe you meant: ['value0','key','value0','value2']. (UNKNOWN_IDENTIFIER), Stack trace (when copying this message, always include the lines below):`

The second failure seems to be failing because :
`2022.08.17 12:12:22.585929 [ 2111713 ] {} <Error> Application: DB::Exception: The local set of parts of table test_33.dst_15 (b56d004b-2894-4e38-9d8e-66d0fb4add01) doesn't look like the set of parts in ZooKeeper: 2.35 thousand rows of 3.10 thousand total rows in filesystem are suspicious. There are 55 uncovered unexpected parts with 2351 rows (10 of them is not just-written with 2092 rows), 0 missing parts (with 0 blocks), 0 covered unexpected parts (with 0 rows).: Cannot attach table `test_33`.`dst_15` from metadata file /var/lib/clickhouse/store/a62/a621c67b-e123-41dd-b7dd-b42937e20f17/dst_15.sql from query ATTACH TABLE test_33.dst_15 UUID 'b56d004b-2894-4e38-9d8e-66d0fb4add01' (`p` UInt64, `k` UInt64, `v` UInt64) ENGINE = ReplicatedMergeTree('/test/01154_move_partition_long_test_33/dst', '15') PARTITION BY p % 10 ORDER BY k SETTINGS index_granularity = 8192
2022.08.17 12:12:31.339349 [ 2112665 ] {} <Error> Application: Caught exception while loading metadata: Code: 231. DB::Exception: The local set of parts of table test_33.dst_15 (b56d004b-2894-4e38-9d8e-66d0fb4add01) doesn't look like the set of parts in ZooKeeper: 2.35 thousand rows of 3.10 thousand total rows in filesystem are suspicious. There are 55 uncovered unexpected parts with 2351 rows (10 of them is not just-written with 2092 rows), 0 missing parts (with 0 blocks), 0 covered unexpected parts (with 0 rows).: Cannot attach table `test_33`.`dst_15` from metadata file /var/lib/clickhouse/store/a62/a621c67b-e123-41dd-b7dd-b42937e20f17/dst_15.sql from query ATTACH TABLE test_33.dst_15 UUID 'b56d004b-2894-4e38-9d8e-66d0fb4add01' (`p` UInt64, `k` UInt64, `v` UInt64) ENGINE = ReplicatedMergeTree('/test/01154_move_partition_long_test_33/dst', '15') PARTITION BY p % 10 ORDER BY k SETTINGS index_granularity = 8192. (TOO_MANY_UNEXPECTED_DATA_PARTS), Stack trace (when copying this message, always include the lines below):`

Please advice if this PR is good to merge ?

**Comment by alexey-milovidov** [comment]
Yes, good to merge.

**Pull request merged**